framework/src/actions/MaterialOutputAction.C

```diff
@@ -340,6 +365,31 @@ MaterialOutputAction::materialOutputHelper<RealVectorValue>(const std::string &
   return names;
 }
 
+template <>
+std::vector<std::string>
+MaterialOutputAction::materialOutputHelper<ADRealVectorValue>(const std::string & property_name,
+                                                              const MaterialBase & material,
+                                                              bool get_names_only)
+{
+  std::array<char, 3> suffix = {{'x', 'y', 'z'}};
+  std::vector<std::string> names(3);
+  for (unsigned int i = 0; i < LIBMESH_DIM; ++i)
+  {
+    std::ostringstream oss;
+    oss << property_name << "_" << suffix[i];
+    names[i] = oss.str();
```

Comment:
I think this should work here, too
```suggestion
    names[i] = property_name + '_' + suffix[i];
```